frontend/src/components/license/licenseUtils.spec.tsx

```diff
@@ -185,4 +279,59 @@ describe('licenseUtils', () => {
       expect(result).toContain('platform=2');
     });
   });
+
+  describe('LicenseNotification Banner', () => {
+    test('render null on routes related to licensing', async () => {
+      const uploadLicenseScreen = renderWithRouter(<LicenseNotification />, {
+        route: '/admin/upload-license',
+      });
+      expect(uploadLicenseScreen.queryByTestId('license-notification')).not.toBeInTheDocument();
+
+      const trialExpiredScreen = renderWithRouter(<LicenseNotification />, {
+        route: '/trial-expired',
+      });
+      expect(trialExpiredScreen.queryByTestId('license-notification')).not.toBeInTheDocument();
+    });
+
+    test('render null if license information is not loaded yet', () => {
+      api.licensesLoaded = undefined;
+      const screen = renderWithRouter(<LicenseNotification />, {
+        route: '/overview',
+      });
+      expect(screen.queryByTestId('license-notification')).not.toBeInTheDocument();
+    });
+
+    test('render information about license that expires in > 15 days but less than 30 days, blue notification banner is displayed', () => {
+      api.licenseViolation = false;
+      api.licenses = [
+        {
+          source: License_Source.REDPANDA_CORE,
+          type: License_Type.ENTERPRISE,
+          expiresAt: getUnixTimestampWithExpiration(28),
+        },
+        {
+          source: License_Source.REDPANDA_CONSOLE,
+          type: License_Type.ENTERPRISE,
+          expiresAt: getUnixTimestampWithExpiration(28),
+        },
+      ];
+      const screen = renderWithRouter(<LicenseNotification />, {
+        route: '/overview',
+      });
+
+      // Check that user is informed about a license that will expire.
+      expect(screen.getByText(/Your Redpanda Enterprise license will expire in 27 days/)).toBeInTheDocument();
```

Comment:
I think this may be sensitive to some timezone changes, we always default to GMT timezone, maybe you can say ```
const DAYS = 28
... expire in ${DAYS - 1}
```